luma/core/bitmap_font.py

```diff
@@ -0,0 +1,632 @@
+            xwidth = max(xwidth, m['xwidth'])
+        self.height = ascent + descent
+        self.width = width
+        self.baseline = ascent
+        self.regular = regular if regular_flag else None
+        self.count = len(self.metrics)
+
+    def combine(self, source_font, characters=None, force=False):
+        """
+        Combine two luma.core.bitmap_fonts
+
+        :param source_font: a luma.core.bitmap_font to copy from
+        :type: luma.core.bitmap_font
+        :param characters: (optional) A list of the characters to transfer from
+            the source_font.  If not provided, all of the characters within
+            the source_font will be transferred.
+        :type characters: str
+        :param force: If set, the source_font can overwrite values that already
+            exists within this font.  Default is False.
+        :type force: bool
+        """
+
+        if characters:
+            for c in characters:
+                if ord(c) in self.mappings and not force:
+                    continue
+                m = source_font.lookup(ord(c))
+                if m is not None:
+                    v = source_font.metrics[m]
+                else:
+                    raise ValueError('{0} is not a valid character within the source font'.format(c))
+                self.metrics.append(v)
+                self.mappings[ord(c)] = len(self.metrics) - 1
+        else:
+            # Copy source values into destination but don't overwrite existing characters unless force set
+            for k, v in source_font.mappings.items():
+                if k in self.mappings and not force:
+                    continue
+                self.metrics.append(source_font.metrics[v])
+                self.mappings[k] = len(self.metrics) - 1
+
+        # Recompute font size metrics
+        self._calculate_font_size()
+
+
+def load(filename):
+    """
+    Load a LUMA.CORE.BITMAP_FONT file.  This function creates a
+    luma.core.bitmap_font object from the given LUMA.CORE.BITMAP_FONT file, and
+    returns the corresponding font object.
+
+    :param filename: Filename of font file.
+    :type filename: str
+    :return: A luma.core.bitmap_font object.
+    :exception OSError: If the file could not be read.
+    :exception SyntaxError: If the file does not contain the expected data
+    """
+    f = bitmap_font()
+    f.load(filename)
+    return f
+
+
+def loads(data):
+    """
+    Load a luma.core.bitmap_font from a string of serialized data.  This function
+    creates a luma.core.bitmap object from serialized data produced from the
+    dumps method and returns the corresponding font object.
+
+    :param data: Serialized luma.core.bitmap_font data.
+    :type data: str
+    :return: A luma.core.bitmap_font object.
+    :exception ValueError: If the data does not a valid luma.core.bitmap_font
+    """
+    f = bitmap_font()
+    f.loads(data)
+    return f
+
+
+def load_pil_font(filename, mappings=None):
```

Comment:
"pil" is a deprecated name, I believe they'd like to call their project 'Pillow' nowadays. So we might as well stick to that as well?